src/kern/exec.c

```diff
@@ -16,44 +16,51 @@
 #include "vmlinux.h"
 #include <bpf/bpf_helpers.h>
 #include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+#include <linux/limits.h>
 
-/*
- * Define a map structure to communicate with the userspace program.
- * The type of map is BPF_MAP_TYPE_PERCPU_ARRAY;
- * it contains one element, found for key = 0
- * and the value is an array with number_of_CPUs elements.
- */
-struct bpf_map_def SEC("maps") output_map = {
-	.type = BPF_MAP_TYPE_PERCPU_ARRAY,
-	.key_size = sizeof(int),
-	.value_size = sizeof(int),
-	.max_entries = 1,
+struct record_sample {
+	int ppid;
+	int pid;
+	int tgid;
+	char name[PATH_MAX];
 };
```

Comment:
Why do you define this record_sample structure twice when you can use the shared process_info structure?

Reply:
forgot to delete the struct from c files :D